core/app/api/v0/c/[communitySlug]/site/[...ts-rest]/route.ts

```diff
@@ -213,6 +215,22 @@ const handler = createNextHandler(
 				};
 			},
 		},
+		searchUsers: async (req, res) => {
+			const { user } = await getLoginData();
+			const users = await getSuggestedUsers({
+				communityId: req.query.communityId,
+				query: {
+					email: req.query.email,
+					firstName: req.query.name,
+					lastName: req.query.name,
+				},
+				limit: req.query.limit,
+			}).execute();
+			return {
+				status: 200,
+				body: users,
+			};
+		},
```

Comment:
related to my comments on the contract (again apologies for not mentioning this earlier), I think we don't need to get the `communityId` from the query, but you can just do `const community = await findCommunityBySlug()`

Ideally we'd do some other kind of permission check (probably just that the user has some kind of membership in the community?), but we can do that after this PR gets in (or if you feel really strongly about it you can use the method implemented in #805, but i don't want to block this pr for that)

Reply:
i think just checking that they're logged in is good enough for now, which is what i've done. eventually we should add a new capability for this i think